packages/dappmanager/src/ethForward/index.ts

```diff
@@ -1,137 +1,79 @@
+    async (_req, hash) => ({ location: "ipfs", hash })
+  );
 
-      // Must change the host from something not *.eth, to prevent the IPFS node
-      // from trying to resolve the domain with DNSLink and causing an error
-      req.headers.host = "dappmanager.dappnode";
+  return (req, res, next): void => {
+    try {
+      const domain = parseEthDomainHost(req);
+      if (domain !== null) {
+        ethForwardHandler(req, res, domain);
+        return;
+      }
 
-      // Note: Must use the promise constructor otherwise proxy.web breaks
-      // due to no proper binding of its underlying 'this'
-      //   Cannot read property 'length' of undefined
-      //   src/node_modules/http-proxy/lib/http-proxy/index.js:72:31
-      await new Promise<http.IncomingMessage>((resolve, reject) => {
-        proxy.web(req, res, { target }, (e: Error | undefined, proxyRes) => {
-          if (!e) return resolve(proxyRes);
-          // Indicates that the host is unreachable.
-          // Usually happens when the node is not running
-          if (e.message.includes("EHOSTUNREACH"))
-            reject(new NodeNotAvailable(e.message, content.location));
-          else reject(new ProxyError(e.message, target));
-        });
-      });
+      const hash = parseIpfsGatewayProxyReqHash(req.url);
+      if (hash !== null) {
+        ipfsGatewayProxydHandler(req, res, hash);
+        return;
+      }
 
-      if (content.location === "ipfs" && params.ETHFORWARD_PIN_ON_VISIT)
-        ipfs.pinAddNoThrow(content.hash);
+      next();
     } catch (e) {
-      /**
-       * Returns the response error HTML. Use function format to make sure
-       * a single HTML is returned to the res stream
-       */
-      function errorToResponseHtml(e: Error, domain?: string): string {
-        logs.debug(`ETHFORWARD Error: ${e.message}`);
-
-        // Not found views
-        if (e instanceof EnsResolverError || e instanceof NotFoundError)
-          return views.notFound(e);
-
-        // Node not available views
-        if (e instanceof EthProviderError) return views.noEth(e);
-        if (e instanceof NodeNotAvailable)
-          if (e.location === "swarm") return views.noSwarm(e);
-          else if (e.location === "ipfs") return views.noIpfs(e);
-
-        // Proxy errors
-        if (e instanceof ProxyError) return views.unknownError(e);
+      next(e);
+    }
+  };
+}
 
-        // Unknown errors, log to error
-        logs.error(`ETHFORWARD Unknown error resolving ${domain}`, e);
-        return views.unknownError(e);
-      }
+function parseEthDomainHost(req: express.Request): string | null {
+  // Check if a request is for a decentralized website, based on their host
+  // - decentral.eth => true
+  // - my.dappmanager.dnp.dappnode.eth => false
+  // - my.dappnode => false
+  const domain = req.headers.host;
+  return typeof domain === "string" &&
+    domain.endsWith(".eth") &&
+    !domain.endsWith("dnp.dappnode.eth")
+    ? domain
+    : null;
```

Comment:
is return null the best approach for that case? wouldn't be better to create another function that returns a boolean if is a domain or not?

Reply:
That's what we where doing before. The issue then is that you have to parse the domain twice. 1) to return true if it's a domain, 2) again to actually do the proxying